modules/generic_worker/templates/run-generic-worker.sh.erb

```diff
@@ -14,6 +14,39 @@ rm -f tasks-resolved-count.txt
 # wrapper script...
 /usr/local/bin/generic-worker "$@" 2>&1 | logger -t generic-worker -s
 
+exitstatus=( ${PIPESTATUS[0]} )
+
+echo $exitstatus
+
+export TASKCLUSTER_CLIENT_ID="<%= @quarantine_client_id %>"
+export TASKCLUSTER_ACCESS_TOKEN="<%= @quarantine_access_token %>"
+
+if [ $exitstatus -eq 69 ]; then
+    if [ -f /Users/cltbld/quarantined ]; then
+        echo "Worker is quarantined"
+    else
+        echo "Quarantine worker"
+        # quarantine the worker
+        /usr/local/bin/quarantine-worker releng-hardware "<%= @worker_type %>" "<%= @worker_group %>" "<%= @hostname %>" 720h5s
```

Comment:
what is the 72 argument, can that be a variable from puppet also (to expose it)

Reply:
> what is the 72 argument

I think 720 hours and 5 seconds (3 days and 5 seconds) - which is how long the worker gets quarantined for. This seems a little low - perhaps it should be an indefinite quarantine (e.g. 100 years), so that somebody has to look into it in order to remove it from quarantine.